packages/serve/src/index.ts

```diff
@@ -27,6 +33,23 @@ export default function serve(...args): void {
         finalArgs['hot'] = webpackArgs._all.hot;
     }
     cli.getCompiler(webpackArgs, core).then((compiler): void => {
-        startDevServer(compiler, finalArgs);
+        const server = startDevServer(compiler, finalArgs);
+
+        if (deprecatedDevServerVersion) {
+            const warning = chalk.yellow.bold(
+                'Using webpack-dev-server v3 with ' +
+                    `${chalk.green('webpack-cli serve')} is deprecated and will lead to ` +
+                    'unexpected behavior. Please upgrade to webpack-dev-server v4.',
+            );
+            server.log.warn(warning);
+
+            // send a warning after the compilation to ensure that the message
+            // reaches the user
+            compiler.hooks.done.tap('serve', () => {
```

Comment:
Why?

Reply:
When the user compiles something, all the compilation stats will be printed and the message will be lost unless it's at the bottom

Reply:
![dev-server-deprecated-warning](https://user-images.githubusercontent.com/22580625/79056017-ab677e80-7c17-11ea-9c5c-97df67ffef38.png)

Reply:
Interesting place, I think we don't need emit warning(s) to stats in such situation

/cc @webpack/cli-team

Reply:
Personally I think it will help the user realize why there might be unexpected behavior.